doc/build-helpers/trivial-build-helpers.chapter.md

```diff
@@ -58,24 +58,115 @@ Variant of `runCommand` that forces the derivation to be built locally, it is no
 This sets [`allowSubstitutes` to `false`](https://nixos.org/nix/manual/#adv-attr-allowSubstitutes), so only use `runCommandLocal` if you are certain the user will always have a builder for the `system` of the derivation. This should be true for most trivial use cases (e.g., just copying some files to a different location or adding symlinks) because there the `system` is usually the same as `builtins.currentSystem`.
 :::
 
-## `writeTextFile`, `writeText`, `writeTextDir`, `writeScript`, `writeScriptBin` {#trivial-builder-writeText}
+## `writeTextFile`, `writeText`, `writeTextDir`, `writeScript`, `writeScriptBin`, `writeShellScript`, `writeShellScriptBin` {#trivial-builder-textwriting}
 
-These functions write `text` to the Nix store. This is useful for creating scripts from Nix expressions. `writeTextFile` takes an attribute set and expects two arguments, `name` and `text`. `name` corresponds to the name used in the Nix store path. `text` will be the contents of the file. You can also set `executable` to true to make this file have the executable bit set.
+`nixpkgs` provides a number of functions that produce derivations which write text into the Nix store.  These include `writeTextFile`, `writeText`, `writeTextDir`, `writeScript`, `writeScriptBin`, `writeShellScript`, and `writeShellScriptBin`, each of which is documented below.
 
-Many more commands wrap `writeTextFile` including `writeText`, `writeTextDir`, `writeScript`, and `writeScriptBin`. These are convenience functions over `writeTextFile`.
+These are useful for creating files from Nix expressions, which may be scripts or non-executable text files, depending on which of the functions is used and the arguments it takes.
+
+The result of each of these functions will be a derivation.  When you coerce the resulting derivation to text, it will evaluate to the *store path*. Importantly, it will not include the destination subpath produced by the particular function.  So, for example, given the following expression:
```

Comment:
```suggestion
Evaluation of a `trivialBuilder` function will return return an attribute set that can be [string interpolated](https://nixos.org/manual/nix/stable/language/string-interpolation.html?highlight=string%20inter#string-interpolation) as a directory path where the the derivation (produced as a side-effect of evaluation) will be realized to. 

Importantly, it this interpolation will not include the destination subpath produced by the particular function.  So, for example, given the following expression:
```

I am not sure to what extent we ought to be describing how derivations work here. Perhaps we can provide a link to the nix reference manual: https://nixos.org/manual/nix/stable/language/derivations.html

But in any case, In Nix lingo:

* functions (more generally, expressions) are "evaluated"
* derivations are "realized" 
* derivations produce attribute sets, which contain an `outPath` value used when interpolating them into strings

I don't know how to explain this well though. So perhaps you can massage my suggested sentence into being clearer, if you find it useful?